odata-parser.d.ts

```diff
@@ -27,7 +27,15 @@ interface GenericPropertyPath<T = any> {
 	name: string;
 	property?: T;
 }
-export type PropertyPath = GenericPropertyPath<PropertyPath>;
+export type PropertyPath = GenericPropertyPath<PropertyPath> & {
+	key?:
+		| BindReference
+		| {
+				[resourceName: string]: BindReference;
+		  };
+	count?: true;
+	options?: { $filter: FilterOption };
```

Comment:
These could be in a union type, butt I preferred keeping it simple for now, so that we can avoid doing `'key' in match` checks in odata-to-abstract-sql.